### DISCUSSION TITLE
GenEnsProd control member settings

### DISCUSSION THREAD

**Discussion by willmayfield** [dtcenter/METplus#1682] [Configuration] [answered]
Hi all! Maybe you have some advice on the control member settings for genensprod..

From documentation:
"The -ctrl command line option specifies an input file for the ensemble control member. The fields specified in the configuration file are read from the control member file. Those fields are included in the computation of the ensemble mean and probabilities but excluded from the ensemble spread."

- do you have a recommendation for how to look up (say, in the code) exactly which genensprod computed outputs exclude the control member when it is set? (We are using NEP, NMEP options and looking at the .nc outputs as well.)

- we are confused about setting the control ID, as that variable is commented out in the basic use case config:
#GEN_ENS_PROD_ENS_MEMBER_IDS =
#GEN_ENS_PROD_CONTROL_ID =
Are these optional? It seems that it will simply pull the name from "GEN_ENS_PROD_CTRL_INPUT_TEMPLATE" to use as the control. But, if the ID option is used, how is that matched with a member from GEN_ENS_PROD_INPUT_TEMPLATE?

Thanks!

Will

**Comment by georgemccabe** [comment, marked as answer]
Hi @willmayfield,

Here is the relevant portion of the gen_ens_prod code that excludes the control member:
https://github.com/dtcenter/MET/blob/1958e02e696283380aef974a298ea98a995b1fc7/met/src/tools/other/gen_ens_prod/gen_ens_prod.cc#L728-L732

GEN_ENS_PROD_ENS_MEMBER_IDS and GEN_ENS_PROD_CONTROL_ID set the ens_member_ids and control_id variables respectively in the GenEnsProd MET config file. These settings are only used if you are reading ensembles that are all found in a single input file. If the ensembles are stored in their own files, then you do not need to use this setting.

More information on these settings and examples can be found in the User's Guide:
https://met.readthedocs.io/en/latest/Users_Guide/gen-ens-prod.html#gen-ens-prod-configuration-file

Please let me know if you have any other questions.

**Reply by willmayfield** [reply]
Thank you George! This was really helpful to understand.
